backend/scripts.js

```diff
@@ -104,34 +104,43 @@ function extractUsersTopTracks (data) {
  * Used to extract data for our use. used in conjuction with GET getMyTopArtists
  * Should pass data.body.items from the above SpotifyAPI request
  * @param {array} data 
- * @returns 
+ * @returns [ array of artist objects, array of array of top genres for every artists] 
  */
-function extractUsersTopArtists (data) {
+function extractUsersTopArtistsAndGenres (data) {
     
-    if (typeof(data) === "undefined") {
+    if (typeof data === "undefined") {
         return "";
     }
 
     let topArtists = [];
+    let topGenres = [];
     // iterate over data and add relevant artist attributes
     for (x of data) {
 
         //if artist name is empty, continue
         if (x.name.length === 0 || x.name.length == null || x.images.length == 0) {
             continue;
-        }
-        
-        topArtists.push({
-            artistName: x.name,
-            artistID: x.id,
-            followerCount: x.followers.total,
-            artistPopularity: x.popularity,
-            imageURL: x.images[0].url,
-            linkURL: x.external_urls.spotify,
-        });
+        } else {
+            topArtists.push({
+                artistName: x.name,
+                artistID: x.id,
+                followerCount: x.followers.total,
+                artistPopularity: x.popularity,
+                imageURL: x.images[0].url,
```

Comment:
Don't know if this will come up, but would be safer if we checked if this array isn't empty before trying to access it

Reply:
this is checked in the above statements on line 121 i believe

Reply:
im blind mbmb